core/src/main/scala/kafka/server/ControllerServer.scala

```diff
@@ -78,18 +78,18 @@ class ControllerServer(
   val awaitShutdownCond = lock.newCondition()
   var status: ProcessStatus = SHUTDOWN
 
-  var linuxIoMetricsCollector: LinuxIoMetricsCollector = null
-  @volatile var authorizer: Option[Authorizer] = null
-  var tokenCache: DelegationTokenCache = null
-  var credentialProvider: CredentialProvider = null
-  var socketServer: SocketServer = null
+  var linuxIoMetricsCollector: LinuxIoMetricsCollector = _
+  @volatile var authorizer: Option[Authorizer] = _
```

Comment:
I just noticed this and maybe it makes sense to initialize this to `None` rather than `_`?

Reply:
I took a quick look and I think we should be able to make that change.

Reply:
Just committed the change, also rebased the PR to latest trunk